File: src/main/python/litcoder/module-1/lab-1/ContiguousArray.py
from typing import List


def solution(arr: List[int]) -> int:
    longest_cont_arr_len = 0
    # check all sub-arrays for longest contiguous sub-array
    for i in range(len(arr)):
        zero_count, one_count = 0, 0
        for j, val in enumerate(arr[i:], i):
            if val == 0:
                zero_count += 1
            else:
                one_count += 1
            if zero_count == one_count:
                longest_cont_arr_len = max(longest_cont_arr_len, j - i + 1)
    return longest_cont_arr_len


inp = list(map(int, input().split(" ")))
print(solution(inp))

File: codes/college/litcoder/module-1/lab-1/ContiguousArray.java
import java.util.Arrays;
import java.util.Scanner;

public class ContiguousArray {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int[] arr = Arrays
                .stream(sc.nextLine().split(" "))
                .mapToInt(Integer::parseInt)
                .toArray();
        System.out.println(solution(arr));

        sc.close();
    }

    private static int solution(int[] arr) {
        int longestContArrLen = 0;

        // check all sub-arrays for all longest contiguous sub-array
        for (int i = 0; i < arr.length; i++) {
            int zeroCount = 0;
            int oneCount = 0;
            for (int j = i; j < arr.length; j++) {
                if (arr[j] == 0) {
                    zeroCount++;
                } else {
                    oneCount++;
                }

                if (zeroCount == oneCount) {
                    longestContArrLen = Math.max(longestContArrLen, j - i + 1);
                }
            }
        }
        return longestContArrLen;
    }
}
